Lib/timeSeries.py

```diff
@@ -84,4 +101,317 @@ def readUnrTxyz2(self, fileName):
         self.time = np.asarray(decYear)
         self.pos = np.stack([x, y, z])
         self.sig = np.stack([sigX, sigY, sigZ])
-        self.covar = np.stack([Cxy, Cyz, Cxz])
+        self.corr = np.stack([Cxy, Cyz, Cxz])
+
+        self.refPos = np.asarray(self.refPos)
+
+    ####################################################################
+    def setRefPosToAvg(self):
+
+        """
+        Reference the time series coordinates to the average position of 
+        the time series. Set the reference position (refPos) to reflect 
+        this translation of coordinates. Does not affect the reference 
+        frame, only shifts the origin for the time series.
+        """
+
+        # this routine should only be used when the coordinates are 
+        # referenced to the frame origin, check that this is the case
+        try:
+            if self.coordType == 'XYZ':
+    
+                # compute average position
+
+                avgX = np.mean(self.pos[0])
+                avgY = np.mean(self.pos[1])
+                avgZ = np.mean(self.pos[2])
+
+                # referenc coodinates to average position
+                self.pos[0] = self.pos[0] - avgX
+                self.pos[1] = self.pos[1] - avgY
+                self.pos[2] = self.pos[2] - avgZ
+
+                # set reference coordinates
+                self.refPos = np.asarray([avgX, avgY, avgZ])
+
+                # set coordinate type to differential coordinates
+                self.coordType = 'dXdYdZ'
```

Comment:
You might consider making these constants so you don't have to type them out all over the place. You also might consider making them subclasses. Not sure what the right fit is.

Reply:
I don't understand what you mean "make these constants". 

self.refPos holds the coordinates of the reference position. The reference position is the position to which all of the coordinates in the time series are referenced. The reference position can change when the coordinates are "transformed".

self.coordType holds a short string that is descriptive of what "type" of coordinates are in the time series. This can also change if the coordinates are transformed.

Reply:
A constant is like a variable that can't be changed. Because it can't be changed, wherever the interpreter sees the constant, it substitutes out the value. For example:

```python
PI = 3.14

a = PI * r * r
```

Is functionally equivalent to 

```python
a = 3.14 * r * r
```

They're useful because they prevent typos and keep things consistent. For example, if you accidentally type `PIE`, Python will yell at you that `PIE` doesn't exist as a constant. If you type `3.13`, nothing will complain because that's a valid float in Python.

In this context here, you're using what we call "magic strings". Magic strings are some random string where, if you see it, it means something, but there isn't any actual structure around it. Somewhere in the code I could set `self.coordType = 'dXdydZ'` and it would take for-fucking-ever to figure out why it wasn't work (subtle typo), but if I set:

```python
DXDYDZ = 'dXdYdZ'
```

At the top of this file and then set `self.coordType = DXDXDZ` on accident, Python will yell at me.

This makes the code more resilient to accidental bugs later.

Reply:
I see. That makes a lot of sense.